pulpcore/app/entrypoint.py

```
@@ -80,12 +80,22 @@ def run(self):
                 self.api_app_status.delete()
 
 
-class PulpcoreApiApplication(BaseApplication):
+class PulpcoreApiApplication(Application):
     def __init__(self, options):
         self.options = options or {}
         super().__init__()
 
+    def init(self, *args, **kwargs):
+        """
+        A hook for setting application-specific configs, which we instead do below in load_config
+        where it's non-overridable.
+        """
+        pass
+
     def load_config(self):
+        # Load default gunicorn configs, including reading from the default config file.
+        super().load_config()
+        # Override with settings that we've specified in the startup script.
```

I'm missing that we prevent the use of `--threads`. And maybe there are more dangerous options?

Do you think we could even reuse the gunicorn commandline parsing at this point?

---

> Is this the reason you moved _away_ from the gunicorn command line parsing that you used to have, because you wanted to enforce `threads=1` and the worker_class? I personally would have just hardcoded those options in the init script and called it a day.

Honestly I don't fully remember. I think is was roughly that.

I think, I preferred failing if `threads` or `worker_class` were specified somewhere. Silently ignoring something a user specified is not too nice. But maybe there's a usecase where a general config file is supposed to provide default options, and we should allow that. We could raise a warning though in case anyone wonders.

---

Okay, check now. I've added warning / error messaging if the user is trying to set something in the config file that will not take effect / cannot be changed.